subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/result/ComponentResultSerializer.java

```diff
@@ -30,6 +31,7 @@
 
 import java.io.IOException;
 import java.util.List;
+import java.util.Set;
 
 public class ComponentResultSerializer implements Serializer<ResolvedGraphComponent> {
```

Comment:
Can you run the full performance tests against this branch?
https://builds.gradle.org/viewType.html?buildTypeId=Gradle_Master_Check_PerformanceTest6_Trigger&branch_Gradle_Master_Check_Stage_ReadyforNightly=master&tab=buildTypeStatusDiv
https://builds.gradle.org/viewType.html?buildTypeId=Gradle_Master_Check_PerformanceTest7_Trigger&branch_Gradle_Master_Check_Stage_ReadyforNightly=master&tab=buildTypeStatusDiv
https://builds.gradle.org/viewType.html?buildTypeId=Gradle_Master_Check_PerformanceTestTestLinux_Trigger&branch_Gradle_Master_Check_Stage_ReadyforNightly=master&tab=buildTypeStatusDiv

The reason I think this is important is that it will show us if we need to adjust our strategy here to serialize less information or skip this information since 99% of the time it'll never be read.